libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -397,36 +397,51 @@ void MaterialBuilder::prepareToBuild(MaterialInfo& info) noexcept {
     info.specularAOSet = mSpecularAOSet;
 }
 
+bool MaterialBuilder::findProperties(filament::backend::ShaderType type,
+        MaterialBuilder::PropertyList& allProperties) noexcept {
+#ifndef FILAMAT_LITE
+    // Use the first permutation to generate the shader code.
+    assert(!mCodeGenPermutations.empty());
+    CodeGenParams params = mCodeGenPermutations[0];
+
+    GLSLTools glslTools;
+    std::string shaderCodeAllProperties = peek(type, params, allProperties);
+    // Populate mProperties with the properties set in the shader.
+    if (!glslTools.findProperties(type, shaderCodeAllProperties, mProperties, params.targetApi,
+            ShaderModel(params.shaderModel))) {
+        return false;
+    }
+    return true;
+#else
+    return false;
+#endif
+}
+
 bool MaterialBuilder::findProperties() noexcept {
     if (mMaterialDomain != MaterialDomain::SURFACE) {
         return true;
     }
 
-#ifndef FILAMAT_LITE
     using namespace filament::backend;
-    GLSLTools glslTools;
 
+#ifndef FILAMAT_LITE
     // Some fields in MaterialInputs only exist if the property is set (e.g: normal, subsurface
     // for cloth shading model). Give our shader all properties. This will enable us to parse and
     // static code analyse the AST.
     MaterialBuilder::PropertyList allProperties;
     std::fill_n(allProperties, MATERIAL_PROPERTIES_COUNT, true);
 
-    // Use the first permutation to generate the shader code.
-    assert(!mCodeGenPermutations.empty());
-    CodeGenParams params = mCodeGenPermutations[0];
-    std::string shaderCodeAllProperties = peek(ShaderType::FRAGMENT, params, allProperties);
-
-    // Populate mProperties with the properties set in the shader.
-    if (!glslTools.findProperties(shaderCodeAllProperties, mProperties, params.targetApi,
-                ShaderModel(params.shaderModel))) {
-        return false;
-    }
+    findProperties(ShaderType::FRAGMENT, allProperties);
+    findProperties(ShaderType::VERTEX, allProperties);
```

Comment:
This was confusing to me at first because it's not immediately clear that `mProperties` won't be reset by multiple calls to `findProperties`. Maybe add a comment here to explain, something like:

```
// Multiple calls to findProperties accumulate the property sets across fragment and vertex shaders in mProperties.
```

Reply:
SG